lib/rest-router.js

```diff
@@ -0,0 +1,37 @@
+module.exports = RestRouter;
+
+var url = require('url');
+var Trie = require('./trie');
+
+function RestRouter(options) {
+  this.trie = new Trie();
+  this.options = options;
+}
+
+//register path and related handler to Trie ds
+RestRouter.prototype.registerPathAndHandlers = function(route, handler) {
+  var path = normalizePath(route.fullPath);
+  var trie = this.trie;
+  trie = trie.add(path, route.verb, handler);
+};
+
+//handle request, match path and if found, invoke handler method
+RestRouter.prototype.handle = function() {
+  var trie = this.trie;
+  return function(req, res, next) {
+    var verb = req.method.toLowerCase() || 'all';
```

Comment:
This is weird. The "all" verb can be used when registering a handler. HTTP request always provide `req.method`.

Please add a unit-test to verify how a handler for `all` verb(s) works.